### ISSUE TITLE
Refactor `nyxt` package definition

### ISSUE THREAD

**Issue by aadcg** [NixOS/nixpkgs#302795]
Nyxt maintainer here.

I had a look at how Nyxt is packaged in Nix and I think it could be simplified greatly.

The package depends on `sbclPackages.nyxt`, but that isn't necessary since the Nyxt sources include the CL dependencies. I propose deleting `sbclPackages.nyxt` and re-writing `pkgs/applications/networking/browsers/nyxt/default.nix` such that it builds Nyxt via `make`.  

Also, note that Nyxt pins specific versions of its CL dependencies. Nix, however, uses their own versions via `lispLibs`. This means that Nix is packaging Nyxt in a way that isn't 100% faithful to the published sources.

In short, I'm advocating for a packaging method where Nyxt manages its CL dependencies by itself, instead of depending on Nix's definition of e.g. `alexandria`. 

This is how it is done in most distributions, [see one example](https://gitlab.archlinux.org/archlinux/packaging/packages/nyxt/-/blob/main/PKGBUILD?ref_type=heads). Nyxt's build process is quite powerful and manages icons or desktop files (i.e. `installPhase` would also be simplified). 

Although I'm not a Nix expert, I could probably give it a try myself. I'm trying to understand if this approach would be accepted. CC @dariof4 @nlewo.

**Comment by aadcg** [comment]
Closed via #321845.
